cli/util/util_test.go

```diff
@@ -63,3 +66,47 @@ func TestUtilOps(t *testing.T) {
 	e.Run(t, "neo-go", "util", "ops", "--hex", "--in", tmp) // hex from file
 	check(t)
 }
+func TestUtilCancelTx(t *testing.T) {
+	e := testcli.NewExecutorSuspended(t)
+
+	w, err := wallet.NewWalletFromFile("../testdata/testwallet.json")
+	require.NoError(t, err)
+
+	tmp := t.TempDir()
+	configPath := filepath.Join(tmp, "config.yaml")
+	cfg := config.Wallet{
+		Path:     testcli.ValidatorWallet,
+		Password: testcli.ValidatorPass,
+	}
+	yml, err := yaml.Marshal(cfg)
+	require.NoError(t, err)
+	require.NoError(t, os.WriteFile(configPath, yml, 0666))
+
+	transferArgs := []string{
+		"neo-go", "wallet", "nep17", "transfer",
+		"--rpc-endpoint", "http://" + e.RPC.Addresses()[0],
+		"--wallet", testcli.ValidatorWallet,
+		"--to", w.Accounts[0].Address,
+		"--token", "NEO",
+		"--from", testcli.ValidatorAddr,
+		"--force",
+	}
+
+	e.In.WriteString("one\r")
+	e.Run(t, append(transferArgs, "--amount", "1")...)
+	line := e.GetNextLine(t)
+	txHash, err := util.Uint256DecodeStringLE(line)
+	require.NoError(t, err)
+
+	_, ok := e.Chain.GetMemPool().TryGetValue(txHash)
+	require.True(t, ok)
+
+	e.Run(t, "neo-go", "util", "canceltx", txHash.StringLE(), "-r", "http://"+e.RPC.Addresses()[0], "--wallet-config", configPath, "--address", testcli.ValidatorAddr)
+	resHash, err := util.Uint256DecodeStringLE(e.GetNextLine(t))
+	require.NoError(t, err)
+
+	_, _, err = e.Chain.GetTransaction(resHash)
+	require.NoError(t, err)
+	e.CheckEOF(t)
```

Comment:
OK, it's not that easy to solve test hanging, because if we don't start chain, then RPC server waits for one of the chain's routines, but at the same time we can't avoid RPC server usage, thus I suggest to add the following code block in the end of the test:
```
	go e.Chain.Run()

	require.Eventually(t, func() bool {
		_, aerErr := e.Chain.GetAppExecResults(resHash, trigger.Application)
		return aerErr == nil
	}, time.Second*2, time.Millisecond*50)
```
This block properly starts the chain and waits until fallback transaction will be accepted. That's exactly what we need to check and this allows to properly shutdown the chain in the end of the test.